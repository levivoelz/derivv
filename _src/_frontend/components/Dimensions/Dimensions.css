.dimensions {
  counter-increment: item;
  margin-bottom: 10px
}

.dimensions:before {
 margin-right: 10px;
 content: counter(item)'.';
 text-align: center;
 display: inline-block;
 font-size: 1.6em;
}
