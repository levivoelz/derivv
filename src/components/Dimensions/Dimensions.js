import React from 'react'
import TextField from 'material-ui/TextField'
import FloatingActionButton from 'material-ui/FloatingActionButton'
import ContentAdd from 'material-ui/svg-icons/content/add'
import ContentClear from 'material-ui/svg-icons/content/clear'
import { TableRow, TableRowColumn } from 'material-ui/Table'

export const Dimensions = (props) => {
  const fieldStyle = {marginRight: 10}
  const {dimensions, onAddClick, onRemoveClick, onChange, lastItem, onlyItem} = props
  const handleRemoveClick = () => { onRemoveClick(dimensions.id) }
  const handleChange = (e) => {
    onChange({
      ...dimensions,
      [e.target.name]: e.target.value
    })
  }

  const renderAddButton = () => (
    <FloatingActionButton mini zDepth={0} onTouchTap={onAddClick}>
      <ContentAdd />
    </FloatingActionButton>
  )

  return (
    <TableRow displayBorder={false}>
      <TableRowColumn>
        <TextField
          floatingLabelText='width in px'
          name='width'
          onChange={handleChange}
          value={dimensions.width}
          type='number' />
      </TableRowColumn>
      <TableRowColumn>
        <TextField
          floatingLabelText='height in px'
          name='height'
          onChange={handleChange}
          value={dimensions.height}
          type='number' />
      </TableRowColumn>
      <TableRowColumn>
        <FloatingActionButton disabled={onlyItem} style={{marginRight: 10}} mini zDepth={0} onTouchTap={handleRemoveClick}>
          <ContentClear />
        </FloatingActionButton>
        {lastItem ? renderAddButton() : ' '}
      </TableRowColumn>
    </TableRow>
  )
}

export default Dimensions
