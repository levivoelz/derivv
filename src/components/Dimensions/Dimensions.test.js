import React from 'react'
import Dimensions from './Dimensions'

const DIMENSIONS = {width: 20, height: 20}

describe('Component: Dimensions', () => {
  it.skip('matches snapshot', () => {
    const component = renderer.create(
      <Dimensions dimensions={DIMENSIONS} />
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('adds another one', () => {
    const onAddClick = jest.fn()
    const component = shallow(<Dimensions
      dimensions={DIMENSIONS}
      lastItem
      onAddClick={onAddClick} />
    )

    component.find('.add-dimensions').simulate('click', { onAddClick })
    expect(onAddClick).toBeCalled()
  })

  it('is not able to add another one', () => {
    const onAddClick = jest.fn()
    const component = shallow(<Dimensions
      dimensions={DIMENSIONS}
      onAddClick={onAddClick} />
    )

    expect(component.find('.add-dimensions')).toHaveLength(0)
  })

  it('removes one', () => {
    const onRemoveClick = jest.fn()
    const component = shallow(<Dimensions
      dimensions={DIMENSIONS}
      onRemoveClick={onRemoveClick} />
    )

    component.find('.remove-dimensions').simulate('click', { onRemoveClick })
    expect(onRemoveClick).toBeCalled()
  })

  it('updates width', () => {
    const onChange = jest.fn()
    const component = shallow(<Dimensions
      dimensions={DIMENSIONS}
      onChange={onChange} />
    )

    component.find('[name="width"]').simulate('change', {target: {value: 1}})
    expect(onChange).toBeCalled()
  })

  it('updates height', () => {
    const onChange = jest.fn()
    const component = shallow(<Dimensions
      dimensions={DIMENSIONS}
      onChange={onChange} />
    )

    component.find('[name="height"]').simulate('change', {target: {value: 1}})
    expect(onChange).toBeCalled()
  })
})
