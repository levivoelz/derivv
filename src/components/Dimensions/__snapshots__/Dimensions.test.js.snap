// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Dimensions matches snapshot 1`] = `
<li
  className="dimensions"
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1"
    onBlur={[Function]}
    onFocus={[Function]}
    required={false}
    style={
      Object {
        "marginRight": 5,
      }
    }
  >
    <label
      className="MuiFormLabel-root-11 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
      data-shrink={true}
      htmlFor={undefined}
    >
      width in px
    </label>
    <div
      className="MuiInput-root-15 MuiInput-formControl-16 MuiInput-inkbar-17 MuiInput-underline-21"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <input
        aria-invalid={false}
        aria-required={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-24 MuiInput-inputSingleline-27"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="width"
        onChange={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="number"
        value={20}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1"
    onBlur={[Function]}
    onFocus={[Function]}
    required={false}
  >
    <label
      className="MuiFormLabel-root-11 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
      data-shrink={true}
      htmlFor={undefined}
    >
      height in px
    </label>
    <div
      className="MuiInput-root-15 MuiInput-formControl-16 MuiInput-inkbar-17 MuiInput-underline-21"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <input
        aria-invalid={false}
        aria-required={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-24 MuiInput-inputSingleline-27"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="height"
        onChange={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="number"
        value={20}
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-37 MuiIconButton-root-30 MuiIconButton-colorPrimary-32 remove-dimensions"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role={undefined}
    style={
      Object {
        "height": 30,
        "padding": 0,
        "width": 30,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label-35"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-39 MuiIconButton-icon-36"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13H5v-2h14v2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-45"
    />
  </button>
   
</li>
`;
